File: Stepometr/TestApp/src/main/java/com/example/testapp/ui/stepometr/pedometr/EveryHourHandler.java
package com.example.testapp.ui.stepometr.pedometr;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.Date;

public class EveryHourHandler extends BroadcastReceiver {

    private static int NOTIFY_ID = 0;

    @Override
    public void onReceive(Context context, Intent intent) {
        PedometerSharedPreferences.WriteCurrentStepsAndTime(context);
    }
}

File: Stepometr/app/src/main/java/com/example/stepometr/stepometr/pedometr/EveryHourHandler.kt
package com.example.stepometr.stepometr.pedometr

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent


class EveryHourHandler : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        PedometerSharedPreferences.writeCurrentStepsAndTime(context)
    }

    companion object {
        private const val NOTIFY_ID = 0
    }
}